meshtastic/interfaces/ble/notifications.py

```
@@ -228,3 +252,480 @@ def _get_callback(self, characteristic: str) -> Callable[[Any, Any], None] | Non
+                report_exception = hook
+                break
+        if report_exception is not None:
+            try:
+                report_exception(error_msg)
+            except Exception:  # noqa: BLE001 - callback error reporting is best effort
+                logger.debug(error_msg, exc_info=True)
+            return
+        logger.debug(error_msg)
+
+    @staticmethod
+    def invoke_safe_execute_compat(
```

Ruff (0.15.6)

[warning] Too many return statements (12 > 6)

(PLR0911)